x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/action/GetRollupCapsAction.java

```diff
@@ -46,13 +48,15 @@ public Response newResponse() {
 
     public static class Request extends ActionRequest implements ToXContent {
         private String indexPattern;
+        private KeyOrder key = KeyOrder.INDEX_PATTERN;
 
-        public Request(String indexPattern) {
+        public Request(String indexPattern, String key) {
```

Comment:
This is a little tricky because if we are allow an index pattern to resolve to concrete indexes then we should allow `IndicesOptions` as well so the user can control how wildcard expansion etc. behaves.  This also introduces a bit of weirdness because the `GetRollupCapsAction` is a cluster action (the name starts with `cluster:` which is correct for when you are specifying the index pattern option because you are effectively asking the job not the indexes but when you are asking for the indexes option the action should really be an indices action (the name should start with `indices:`). I wonder if this actually needs to be two different APIs? or whether there is a way to make this always an `indices:` action?

Reply:
Hm, maybe two endpoints is best.

Reply:
Durp, hit enter too soon.

Right now I don't think it's a problem, since the options are:

1. "Index pattern" specified in the job, which we are not actually resolving as index patterns.  These are just checking string equality against the job's defined index pattern.

2. The new behavior introduced in this PR, which requests the rollup job's concrete index.  This _does not_ support patterns at the moment, only a single, explicit index name.

So the PR as it stands should be ok.  But Kibana has said they would like to be able to specify patterns with the concrete rollup index (`rollup_storage_*`), which would circle back to the concerns you've raised.